From 31c0f1cb59ec0c36c3ca718a5fab89416b986f2b Mon Sep 17 00:00:00 2001
From: Filip Lobpreis <f.lobpreis@gmail.com>
Date: Mon, 5 Dec 2022 09:33:58 +0100
Subject: [PATCH 2/3] pt2

Signed-off-by: Filip Lobpreis <f.lobpreis@gmail.com>
---
 main.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/main.cpp b/main.cpp
index a56bd62..889a237 100644
--- a/main.cpp
+++ b/main.cpp
@@ -58,10 +58,15 @@ int main(int argc, char *argv[])
 	// std::cout << std::endl;
 	for(auto proc : procedure) {
 		// printTops(data);
+		std::stack<char> help;
 		for (size_t i = 0; i < proc.move; i++) {
-			data[proc.to-1].push(data[proc.from-1].top());
+			help.push(data[proc.from-1].top());
 			data[proc.from-1].pop();
 		}
+		for (size_t i = 0; i < proc.move; i++) {
+			data[proc.to-1].push(help.top());
+			help.pop();
+		}
 	}
 
 	for(auto var : data) {
-- 
2.34.1

